src/ast_editor.ml

```diff
@@ -330,10 +336,16 @@ let open_pp_doc instance ~document =
   match fetch_pp_code ~document with
   | Error e -> Promise.return (Error e)
   | Ok pp_pp_str ->
+    let file_name =
+      match Stdlib.Filename.extension (TextDocument.fileName document) with
+      | ".re" -> (String.chop_suffix_exn ~suffix:".re" (TextDocument.fileName document) ^ ".ml")
+      | ".rei" -> (String.chop_suffix_exn ~suffix:".rei" (TextDocument.fileName document) ^ ".mli")
+      | _ -> TextDocument.fileName document
+    in
```

Comment:
In the future instead of providing a .ml for pp document we could convert ocaml code to reasonml.